File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include <iostream>
#include <vector>
#include <queue>
#include <algorithm>

using namespace std;

int N, M, V;

void dfs(int x, vector<int> graph[], bool visited[]){

    visited[x] = true;

    cout << x << " ";

    for(int i=0; i<graph[x].size(); i++){

        int y = graph[x][i];
        
        if (visited[y] == false){
            dfs(y, graph, visited);
        }
    }
}

void bfs(int x, vector<int> graph[], bool visited[]){
    queue<int> q;
    q.push(x);
    visited[x] = true;

    while(!q.empty()){
        int y = q.front();
        cout << y << " ";
        q.pop();

        for(int i=0; i<graph[y].size(); i++){
            int node = graph[y][i];
            if (visited[node] == false){
                q.push(node);
                visited[node] = true;
            }
        }
    }

}

int main(){
    cin >> N >> M >> V;

    vector<int> graph[N+1];
    bool visited[N+1] = {false};
    bool visited1[N+1] = {false};

    for(int i=0; i<M; i++){
        int num1, num2;
        cin >> num1 >> num2;
        graph[num1].push_back(num2);
        graph[num2].push_back(num1);
    }

    for(int i=0; i < N+1; i++){
        sort(graph[i].begin(), graph[i].end());
    }

    //bfs(V, graph, visited);
    dfs(V, graph, visited);
    
    cout << "\n";

    bfs(V, graph, visited1);

}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Main {

    static int n, m, v;
    static ArrayList<ArrayList<Integer>> graph = new ArrayList<>();
    static boolean[] visited;
    
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());
        v = Integer.parseInt(st.nextToken());

        for(int i=0; i<n+1; i++){
            graph.add(new ArrayList<>());
        }

        for(int i=0; i<m; i++){
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());  
            int b = Integer.parseInt(st.nextToken());  

            graph.get(a).add(b);
            graph.get(b).add(a);
        }

        for(int i=1; i<n+1; i++){
            Collections.sort(graph.get(i));
        }

        visited = new boolean[n+1];
        dfs(v);

        System.out.println();

        visited = new boolean[n+1];
        bfs(v);
    }

    static void dfs(int start){
        
        visited[start] = true;
        System.out.print(start + " ");

        for(int next : graph.get(start)){
            if(visited[next]) continue;
            dfs(next);
        }
    }

    static void bfs(int start){
        Queue<Integer> q = new LinkedList<>();
        visited[start] = true;
        q.add(start);
        System.out.print(start + " ");

        while(!q.isEmpty()){
            int now = q.poll();

            for(int next : graph.get(now)){
                if(visited[next]) continue;
                visited[next] = true;
                System.out.print(next + " ");
                q.add(next);
            }
        }
    }
}